crates/lang/macro/src/lib.rs

```diff
@@ -665,6 +666,149 @@ pub fn trait_definition(attr: TokenStream, item: TokenStream) -> TokenStream {
     trait_def::analyze(attr.into(), item.into()).into()
 }
 
+/// Prepares the type to be fully compatible and usable with the storage.
+/// It implements all necessary traits and calculates the storage key for types.
+/// Packed types don't have a storage key, but non-packed types
+/// (like `Mapping`, `Lazy` etc.) require calculating the storage key during compilation.
+///
+/// Consider annotating structs and enums that are intented to be a part of
+/// the storage with this macro. If the type is packed then the usage of the
+/// macro is optional.
+///
+/// If the type is non-packed it is best to rely on automatic storage key
+/// calculation. The storage key can also be specified manually with the
+/// generic `KEY` parameter though.
+///
+/// The macro should be called before `derive` macros because it can change the type.
+///
+/// All required traits can be:
+/// - Derived manually via `#[derive(...)]`.
+/// - Derived automatically via deriving of `scale::Decode` and `scale::Encode`.
+/// - Derived via this macro.
+///
+/// # Example
+///
+/// ## Trait implementation
+///
+/// ```
```

Comment:
I'm having a hard time following all these docs. There's a lot of variance with the examples and there's no real indication as to when each case would apply.

Plus, only a couple show how to actually use `storage_item` (which is what the main focus of these docs should be).